app/services/billing_entities/update_service.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+module BillingEntities
+  class UpdateService < BaseService
+    Result = BaseResult[:billing_entity]
+
+    def initialize(billing_entity:, params:)
+      @billing_entity = billing_entity
+      @params = params
+
+      super(nil)
+    end
+
+    def call
+      billing_entity.email = params[:email] if params.key?(:email)
+      billing_entity.legal_name = params[:legal_name] if params.key?(:legal_name)
+      billing_entity.legal_number = params[:legal_number] if params.key?(:legal_number)
+      if params.key?(:tax_identification_number)
+        billing_entity.tax_identification_number = params[:tax_identification_number]
+      end
+      billing_entity.address_line1 = params[:address_line1] if params.key?(:address_line1)
+      billing_entity.address_line2 = params[:address_line2] if params.key?(:address_line2)
+      billing_entity.zipcode = params[:zipcode] if params.key?(:zipcode)
+      billing_entity.city = params[:city] if params.key?(:city)
+      billing_entity.state = params[:state] if params.key?(:state)
+      billing_entity.country = params[:country]&.upcase if params.key?(:country)
+      billing_entity.default_currency = params[:default_currency]&.upcase if params.key?(:default_currency)
+      billing_entity.document_numbering = params[:document_numbering] if params.key?(:document_numbering)
+      billing_entity.document_number_prefix = params[:document_number_prefix] if params.key?(:document_number_prefix)
+      billing_entity.finalize_zero_amount_invoice = params[:finalize_zero_amount_invoice] if params.key?(:finalize_zero_amount_invoice)
+
+      billing = params[:billing_configuration]&.to_h || {}
+      billing_entity.invoice_footer = billing[:invoice_footer] if billing.key?(:invoice_footer)
+      billing_entity.document_locale = billing[:document_locale] if billing.key?(:document_locale)
+
+      # NOTE: handle eu tax management for billing_entity
+      handle_eu_tax_management(params[:eu_tax_management]) if params.key?(:eu_tax_management)
+
+      if License.premium? && billing.key?(:invoice_grace_period)
+        # for now we won't update related invoices to the billing_entity from this service
+        BillingEntities::UpdateInvoiceGracePeriodService.call(
+          billing_entity:,
+          grace_period: billing[:invoice_grace_period]
+        )
+      end
+
+      if params.key?(:net_payment_term)
+        BillingEntities::UpdateInvoicePaymentDueDateService.call(
+          billing_entity:,
+          net_payment_term: params[:net_payment_term]
+        )
+      end
+
+      assign_premium_attributes
+      handle_base64_logo if params.key?(:logo)
+
+      billing_entity.save!
+
+      # I guess we dont need this...
+      # ApiKeys::CacheService.expire_all_cache(billing_entity)
+
```

Comment:
if we don't need this, just delete the code. It could be added if we decide to manage api keys at billing entity level.

Reply:
🚀